### PR TITLE
Lattices and catwalks now use generic icon smoothing.

### PR DESCRIPTION
**Pull request by xxalpha**
Also added warnings for overlapping lattices and lattices on non-space turf.

Sprite work by @Wjohn

Edit: Expanded smooth code a bit, added a mode to smooth with all subtypes instead of exact types. This is useful for lattice because it should be smoothing with all walls and all floors.

### CODE REVIEW

**Comment by optimumtact** [comment]
:+1: nice work

File: code/game/objects/structures/lattice.dm [hunk 1]
```diff
@@ -93,7 +78,9 @@
 /obj/structure/lattice/catwalk
 	name = "catwalk"
 	desc = "A catwalk for easier EVA manuevering and cable placement."
-	icon_state = "catwalkfull"
+	icon = 'icons/obj/smooth_structures/catwalk.dmi'
+	icon_state = "catwalk"
+	canSmoothWith = null
```
Comment (Anonus) [inline code/game/objects/structures/lattice.dm]: Won't this make catwalks not smooth with eachother or am I missing something?

Comment (xxalpha) [inline code/game/objects/structures/lattice.dm]: You are, when that list var is null the atom will smooth only with itself.

Comment (Anonus) [inline code/game/objects/structures/lattice.dm]: Well that sure is confusing.

File: code/game/objects/structures/lattice.dm [hunk 2]
```diff
@@ -1,37 +1,40 @@
...
 	..()
+	var/atom/locsave = loc
 	if(!(istype(src.loc, /turf/space)))
+		spawn(10)
+			warning("Lattice mapped on non-space turf at ([locsave.x],[locsave.y],[locsave.z])")
 		qdel(src)
 	for(var/obj/structure/lattice/LAT in src.loc)
 		if(LAT != src)
+			spawn(10)
+				warning("Overlapped lattices at ([locsave.x],[locsave.y],[locsave.z])")
```
Comment (Aranclanos) [inline code/game/objects/structures/lattice.dm]: why the spawns?

Comment (xxalpha) [inline code/game/objects/structures/lattice.dm]: If I don't use spawn the warnings never get output.

**Comment by Dorsisdwarf** [comment]
Ready 4 merge?

**Comment by xxalpha** [comment]
Yes

**Pull request merged**